### DISCUSSION TITLE
Passing JSON into SQL helper

### DISCUSSION THREAD

**Discussion by sebpowell** [prisma/prisma#17932] [Q&A]
Hello everyone – hoping someone might be able to help me as I've been stuck for a few hours. We are using a raw query to access a function within our Postgres database, in the following way:

```TS
await this.prismaService.$queryRaw`${query}`
```

The query itself looks like this:

```TS
const query = Prisma.sql`
 SELECT * FROM function(current_date, cast(${quantity} as int), '${json}')
`;
```

`quantity` in this case is a number, which is passed in to the sql helper. This works without any problems.

We've now added an additional, parameter, which in the above example is called json and it's an array of JSON objets that looks something like this:

```TS
[
  {
    "size": 24,
  },
  {
    "size": 12
  }
]
```

I tried passing it in by hardcoding it as follows, and it worked:

```TS
const json = Prisma.sql`[{size: 24}, {size: 12}]`
```

```TS
const query = Prisma.sql`
 SELECT * FROM function(current_date, cast(${quantity} as int), '${json}')
`;
```

But if I do something like this:

```TS
const inputJson = [
  {
    size: 24,
  },
  {
    size: 12
  }
]

const json =  Prisma.sql`${inputJson}`

const query = Prisma.sql`
 SELECT * FROM function(current_date, cast(${quantity} as int), '${json}')
`;
```

I get the following error:

```TS
Invalid `prisma.$queryRaw()` invocation:

Raw query failed. Code: `22P02`. Message: `db error: ERROR: invalid input syntax for type json
DETAIL: Token "$" is invalid.`
```

And I need to do something like this, as hardcoding the values obviously isn't much good.

I hope I've made this clear. Any ideas where I might be going wrong?

**Comment by nurul3101** [comment]
Hey @sebpowell 👋 

Which version of Prisma are you using? Are you using a version above 4.0.0? We shipped some improvements regarding queryRaw in 4.0.0 and above.

**Comment by sebpowell** [comment]
@nurul3101 – thanks for the quick response! Sorry I should have specified; we're using 4.9.0, so I'm guessing we should in theory have access to these improvements?

**Reply by sebpowell** [reply]
@nurul3101 I was just wondering if you happen to have any other thoughts on the above? 

We've been trying a whole bunch of different things, but no luck so far. Any pointers would be greatly appreciated. Thank you!

**Reply by nurul3101** [reply]
Looks like it maybe a bug from our end. Can you share the function that you are invoking? It would help if I could replicate this.

**Reply by sebpowell** [reply]
@nurul3101 – thanks! Just to check, do you want me to share the actual function that we're calling within the database, or just what we're doing within the actual application?

**Reply by nurul3101** [reply]
The actual function that you have defined in the database, I would define the same function in my local database and investigate this

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
